crates/engine/tree/src/persistence.rs

```diff
@@ -0,0 +1,80 @@
+use crate::tree::ExecutedBlock;
+use futures::ready;
+use reth_primitives::B256;
+use reth_provider::ProviderFactory;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::sync::mpsc::Receiver;
+
+// TODO: design questions:
+// * to support "unwinds" or not? ie, what do we do when we get a request to persist a chain that is
+// overlapping with the one on disk
+/// Writes parts of reth's in memory tree state to the database.
+pub struct Persistence<DB> {
+    /// The db / static file provider to use
+    provider: ProviderFactory<DB>,
+    // TODO: handles for pushing requests
+    /// Incoming requests to persist stuff
+    incoming: Receiver<PersistenceAction>,
```

Comment:
do we expect this to be a spawned service that dispatches the actions?

if not then this perhaps doesn't need the receiver and instead controls only the spawned threads that do the IO

Reply:
yeah, this would be spawned, are you suggesting just passing the receiver to the spawned threads then?

Reply:
> yeah, this would be spawned,

then this makes sense, but we also need to spawn the actual IO work to a dedicated thread